addons/point_of_sale/static/src/app/models/product_product.js

```diff
@@ -215,10 +215,26 @@ export class ProductProduct extends Base {
         if (!this._archived_combinations) {
             return false;
         }
+        const excludedPTAV = new Set();
         for (const archivedCombination of this._archived_combinations) {
             const ptavCommon = archivedCombination.filter((ptav) =>
                 attributeValueIds.includes(ptav)
             );
+            if (ptavCommon.length === attributeValueIds.length) {
+                // all attributes must be disabled from each other
+                archivedCombination.forEach((ptav) => excludedPTAV.add(ptav));
+            } else if (ptavCommon.length === attributeValueIds.length - 1) {
+                // In this case we only need to disable the remaining ptav
+                const disablePTAV = archivedCombination.find(
+                    (ptav) => !attributeValueIds.includes(ptav)
+                );
+                excludedPTAV.add(disablePTAV);
+            }
+            this.attribute_line_ids.forEach((attribute_line) => {
+                attribute_line.product_template_value_ids.forEach((ptav) => {
+                    ptav["excluded"] = excludedPTAV.has(ptav.id);
+                });
+            });
             if (ptavCommon.length === attributeValueIds.length) {
                 return true;
             }
```

Comment:
```suggestion
            });
            if (ptavCommon.length === attributeValueIds.length) {
                archivedCombination = true;
            }
```

and return `archivedCombination`